### PR TITLE
Remove unixtimestamp type

### PR DESCRIPTION
**Pull request by dpsutton**
Remove almost all the remaining usages of the coercions as semantic types.

highlighting one of the commit messages:

`:type/UNIXTimestampSeconds and type/ISO8601DateTimeString out of type`

remove the coercions from the type hierarchy. This brought up a
strange issue that has been present for a while: all liquidbase
migrations run and then all data migrations run. They are not
interleaved. This allows for the following scenario:

- data migration migrates all X values to Y in version 26
- liquibase migration migrates all Y values to Z in version 28

But these are not run in version order, but all liquibase migrations
are run and then all data migrations are run. If you were on an old
version for a while, you could end up with Y values even though the
liquibase migration which comes "after" the data migration turns all Y
values into Z values.

This impacts this change because a data migration in this way:
- liquibase changesets 287, 288, 289, and 290 remove all 'type/UNIX*'
and 'type/ISO8601*' semantic types. These were in 0.39
- data migration `migrate-field-types` added in 0.20.0 was looking for
special_type's of "timestamp_milliseconds" and migrating them to
"type/UNIXTimestampMilliseconds".

Since the liquibase runs before the migrate-field-types, it's possible
for a 'type/UNIX*' semantic type to reappear. And since these were
removed from the type system it would fail validation and blow up. In
this case it actually can't happen since the field its attempting to
migrate (special_type) no longer exists. But since the migrations are
not interleaved this problem still exists.

### CODE REVIEW

File: shared/src/metabase/types.cljc [hunk 1]
```diff
@@ -106,18 +106,6 @@
...
-(derive :type/UNIXTimestampSeconds :type/UNIXTimestamp)
-(derive :type/UNIXTimestampMilliseconds :type/UNIXTimestamp)
-(derive :type/UNIXTimestampMicroseconds :type/UNIXTimestamp)
-
-(derive :type/TemporalString :type/Text)
-(derive :type/TemporalString :type/DateTime)
-(derive :type/ISO8601DateTimeString :type/TemporalString)
-(derive :type/ISO8601DateString :type/TemporalString)
-(derive :type/ISO8601TimeString :type/TemporalString)
-
```
Comment (dpsutton) [inline shared/src/metabase/types.cljc:120]: due to the weirdness of the migrations i'm thinking about leaving these in the type hierarchy for a bit but not exposing them to the frontend version of the type hierarchy. Meaning the backend wouldn't blow up and the frontend wouldn't be able to select them as a new choice.

**Comment by camsaul** [comment]
@dpsutton why don't you just remove the data migration and rewrite it as a Liquibase migration -- or even add it to one of the new ones you created for the type system overhaul? That migration always should have been a Liquibase one in the first place IMO.

File: shared/src/metabase/types.cljc [hunk 2]
```diff
@@ -104,6 +104,10 @@
 ;; different from `java.time.OffsetDateTime`;
 (derive :type/Instant :type/DateTimeWithLocalTZ)
 
+
+;; TODO: remove these. They shouldn't be allowed as semantic types. But migrations use them targeting version
+;; 0.20. Not clear if we should remove the tests or keep these forever
+
```
Comment (camsaul) [inline shared/src/metabase/types.cljc:110]: I don't want to end up keeping things forever -- our code is complicated enough without a bunch of legacy cruft hanging around. I vote to remove them

File: test/metabase/sync/sync_metadata/fields/fetch_metadata_test.clj [hunk 3]
```diff
@@ -39,8 +39,8 @@
              {:name              "ts"
               :database-type     "BIGINT"
               :base-type         :type/BigInteger
-              :effective-type    :type/BigInteger
-              :semantic-type     :type/UNIXTimestampMilliseconds
+              :effective-type    :type/DateTime
```
Comment (camsaul) [inline test/metabase/sync/sync_metadata/fields/fetch_metadata_test.clj:42]: I think these should technically come back with effective type `:type/Instant` now but it probably makes zero difference here

**Review by camsaul — CHANGES_REQUESTED** [review]
I'd much rather we just remove the old types entirely and add case statements to deal with pre-20 instances to one of your recent Liquibase migrations to migrate the legacy-legacy types than leave crufty code around forever. I think the duplicate types there would cause confusion that could lead to bugs in the future -- people might not know not to use `type/UNIXTimestampSeconds` and accidentally use it in the future

File: src/metabase/db/data_migrations.clj [hunk 4]
```diff
@@ -126,9 +126,9 @@
 ;;; |                                                NEW TYPE SYSTEM                                                 |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
-;; this is the old new type system. as of v30 these are migrated from a field named special_type to
+;; this is the old new type system. as of v30 these are migrated from a field named semantic_type to
 ;; semantic_type. These migrations target the world before this migration so they are left as is. The migrations talk
```
Comment (camsaul) [inline src/metabase/db/data_migrations.clj:130]: They're migrated from `sematic_type` to... `semantic_type`? I think you mean v39 too. Not v30

**Review by camsaul — APPROVED** [review]
This looks good to me for now. I think we should probably go in and convert the rest of that Clojure data migration into a Liquibase migration as a tech debt thing in the future (not as part of this PR) along with any other Clojure  data migrations that we can convert to Liquibase ones. Just so we don't run into order-related weirdness in the future. If it was a Liquibase migration in the first place we wouldn't have to deal with all this drama

**Pull request merged**